### PR TITLE
Feature/copy integrator

### PR DESCRIPTION
**Pull request by dymczykm**

### CODE REVIEW

File: voxblox/CMakeLists.txt [hunk 1]
```diff
@@ -54,9 +54,22 @@ cs_add_library(${PROJECT_NAME}
...
 target_link_libraries(${PROJECT_NAME} ${PROTOBUF_LIBRARIES})
 
-#########
-# TESTS #
-#########
+##############
+# BENCHMARKS #
+##############
+find_package(benchmark_helpers REQUIRED)
+add_benchmark(bm_cast_ray test/benchmark_cast_ray.cc)
```
Comment (schneith) [inline voxblox/CMakeLists.txt:61]: with this cmake macro it is now possible to run benchmark just as gtests with:
make run_benchmarks: runs all
make run_benchmarks_NAME: runs only this benchmark
make benchmarks: builds them all
...

File: benchmark_helpers/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+cmake_minimum_required(VERSION 2.8.3)
```
Comment (mfehr) [inline benchmark_helpers/CMakeLists.txt:1]: @schneith shouldn't this be part of benchmark_catkin or a dependency of it?

Comment (schneith) [reply benchmark_helpers/CMakeLists.txt:1]: My idea was to create an own repo "benchmark_helpers" where we place that stuff. But for now it just left it in the voxblox repo until it is proven to work. Wdyt?

File: benchmark_helpers/cmake/benchmark-helpers.cmake.em [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  # Add include directory if available.
+  set(${PROJECT_NAME}_LOCAL_INCLUDE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/include)
+  if(NOT IS_DIRECTORY ${${PROJECT_NAME}_LOCAL_INCLUDE_DIR})
+    set(${PROJECT_NAME}_LOCAL_INCLUDE_DIR)
+  endif()
+  include_directories(${${PROJECT_NAME}_LOCAL_INCLUDE_DIR} ${catkin_INCLUDE_DIRS})
+  
+  # Build the test (but not in target all).
+  cs_add_executable(${target} ${ARGN})
```
Comment (schneith) [inline benchmark_helpers/cmake/benchmark-helpers.cmake.em:14]: rm cs_

File: benchmark_helpers/cmake/benchmark-helpers.cmake.em [hunk 4]
```diff
@@ -0,0 +1,45 @@
+function(add_benchmark target)
+  if(NOT DEFINED CMAKE_RUNTIME_OUTPUT_DIRECTORY)
+    message(FATAL_ERROR "add_executable_with_benchmark() must be called after catkin_package() so that default output directories for the executables are defined")
```
Comment (schneith) [inline benchmark_helpers/cmake/benchmark-helpers.cmake.em:3]: add_benchmark

File: benchmark_helpers/cmake/benchmark-helpers.cmake.em [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    add_custom_command(
+    	OUTPUT ${OUTPUT_FILE}
+    	COMMAND ${cmd} 
+    	DEPENDS ${target} 
+    	WORKING_DIRECTORY ${ARG_WORKING_DIRECTORY}
+    )
+    add_custom_target(run_benchmarks_${target} DEPENDS ${OUTPUT_FILE})
+    
+    # Add to benchmark meta-target.
+  	if(NOT TARGET run_benchmarks)
```
Comment (schneith) [inline benchmark_helpers/cmake/benchmark-helpers.cmake.em:40]: indent

File: benchmark_helpers/CMakeLists.txt (hunk 2 above)
Comment (schneith) [reply benchmark_helpers/CMakeLists.txt:1]: hm, having it in benchmark_catkin would be easier probably. i'll move it there

File: voxblox/CMakeLists.txt (hunk 1 above)
Comment (mfehr) [reply voxblox/CMakeLists.txt:61]: To run directly from catkin tools:

```
catkin build <package> -v --no-deps --make-args run_benchmarks
```

**Review by mfehr — APPROVED** [review]

**Pull request merged**